src/main/java/com/sda/games/checkers/model/Game.java

```diff
@@ -1,71 +1,155 @@
 package com.sda.games.checkers.model;
 
+import lombok.*;
+
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
 public class Game {
-    static String whitePlayerName;
-    static String blackPlayerName;
 
-    private Player[] players;
+    private List<Player> players;
     private Board board;
-    private Player currentTurn;
+    private Player currentTurnPlayer;
     private GameStatus status;
     private List<Move> movesPlayed;
+    private Scanner scanner = new Scanner(System.in);
+
+    public void newGame() {
+        System.out.println("Enter White Player name:");
+        Player whitePlayer = new Player(1, Player.whitePlayerName = scanner.nextLine(), true);
+        System.out.println("Enter Black Player name:");
+        Player blackPlayer = new Player(2, Player.blackPlayerName = scanner.nextLine(), false);
 
-    public Game(Player[] players) {
-        this.players = players;
+
+        initialize(whitePlayer, blackPlayer);
+        getBoard().printBoard();
     }
 
     public void initialize(Player player1, Player player2) {
-        players[0] = player1;
-        players[1] = player2;
+        this.players = new ArrayList<>();
+        players.add(player1);
+        players.add(player2);
 
+        this.board = new Board();
         board.resetBoard();
 
-        if (player1.getWhite()) {
-            this.currentTurn = player1;
+        if (player1.isWhite()) {
+            this.currentTurnPlayer = player1;
         } else {
-            this.currentTurn = player2;
+            this.currentTurnPlayer = player2;
         }
 
+        this.movesPlayed = new ArrayList<>();
         movesPlayed.clear();
     }
 
-    public boolean isEnd()
-    {
-        return this.getStatus() != GameStatus.ACTIVE;
-    }
+    public boolean makeMove() throws Exception {
 
-    public GameStatus getStatus() {
-        return status;
-    }
+        Player currentPlayer = getCurrentTurnPlayer();
 
-    public void setStatus(GameStatus status) {
-        this.status = status;
-    }
+        System.out.println(getCurrentTurnPlayer().getName() + " move.");
+
+        Pattern movePattern = Pattern.compile("[a-hA-H][1-8]");
+
+        String startSpotXY = "";
+        char charStartY;
+        char charStartX;
+        int startX;
+        int startY;
+        String endSpotXY = "";
+        char charEndX;
+        char charEndY;
+        int endX;
+        int endY;
+        Piece sourcePiece;
+
+        // Choosing checker to move
+        while (true) {
```

Comment:
ogolnie funkcje `while (true)` sa bardzo niemilo widziane, zawsze powinienes dac jakis warunek stopu, no ale pewnie nie ma tu calej funkcjonalnosci wiec liczne na to ze w przyszlosci to sie zmieni :)